regression-tests/test-results/pure2-deducing-pointers-error.cpp2.output

```diff
@@ -5,5 +5,8 @@ pure2-deducing-pointers-error.cpp2(20,9): error: = - pointer assignment from nul
 pure2-deducing-pointers-error.cpp2(21,9): error: += - pointer assignment from null or integer is illegal
 pure2-deducing-pointers-error.cpp2(25,9): error: = - pointer assignment from null or integer is illegal
 pure2-deducing-pointers-error.cpp2(29,9): error: = - pointer assignment from null or integer is illegal
+pure2-deducing-pointers-error.cpp2(31,11): error: ++ - pointer arithmetic is illegal - use std::span or gsl::span instead
+pure2-deducing-pointers-error.cpp2(33,8): error: = - pointer assignment from null or integer is illegal
   ==> program violates lifetime safety guarantee - see previous errors
+  ==> program violates bounds safety guarantee - see previous errors
```

Comment:
Now I'm getting a legit error too:
```output
/home/johel/Documents/C++/Repos/Waarudo/sources/waarudo/sfml/sfml.cpp2...
sfml.cpp2(86,67): error: [ - pointer arithmetic is illegal - use std::span or gsl::span instead
  ==> program violates bounds safety guarantee - see previous errors
```
```Cpp2
  export vertex_ptr: @basic_value <Const: bool> type = {
    vertex: * maybe_const<Const, sf::Vertex> = std::nullptr_t();
    // ...
    operator[]: (this, i: std::ptrdiff_t) -> vertex_ptr   = vertex[i]&;
    // ...
  }
```
I was aware of this issue.
I remember thinking that doubling the size seemed like an overkill for a "pointer".
But now I remembered that there's `std::span<T, N>` where `N != std::dynamic_extent`.
Also, the error message made me realize that what I really want is a `vertex_span`.
Thanks to the error message pointing that out, it is now evident to me looking at `operator[]`.

Reply:
I'm not 100% clear whether this is bad or good... were the cppfront diagnostics a problem, or helpful?

Reply:
Definitely good for now.
Some day, I might care about performance enough,
and the extra abstractions might turn out to be a bottle neck.
But I'm not ready to measure.